src/main/java/li/cil/oc2/api/inet/TransportLayer.java

```diff
@@ -0,0 +1,60 @@
+package li.cil.oc2.api.inet;
+
+/**
+ * Transport TCP/IP layer interface.
+ * <p>
+ * There is a default transport layer implementation that uses provided {@link SessionLayer} implementation
+ * (see {@link SessionLayerInternetProvider} for more information).
+ */
+public interface TransportLayer {
+
+    /**
+     * The value of this constant should be returned by {@link TransportLayer#receiveTransportMessage(TransportMessage)}
+     * method if no data is arrived.
+     */
+    byte PROTOCOL_NONE = 0;
+
+    /**
+     * The value of this constant should be returned by {@link TransportLayer#receiveTransportMessage(TransportMessage)}
+     * method if an ICMP message is arrived.
+     */
+    byte PROTOCOL_ICMP = 1;
```

Comment:
Just some random thoughts from an IPv6 nerd you're missing a `PROTOCOL_ICMP6` which should have a value of 58 in order to deal with IPv6 ping